examples/math/gsm8k/gsm8k_assertions.ipynb

```diff
@@ -0,0 +1,653 @@
+        "2.   Optimize the DSPy program with in-context learning and prompt tuning.\n",
+        "\n"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "NckBcbZN9p6y"
+      },
+      "outputs": [],
+      "source": [
+        "# Set up your API key for OpenAI\n",
+        "import os\n",
+        "os.environ[\"OPENAI_API_KEY\"]=\"Paste your key here\""
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "ghG4tCe3e-kQ"
+      },
+      "source": [
+        "### Step -1. **Installing Cache and DSPy** (Run the collapsed cells)\n",
+        "\n",
+        "The cells are collapsed by default. Running the following cells will set up cache and install all dependencies and DSPy.\n",
+        "\n",
+        "The first cell ensures all the following LM calls in this notebook will be using the cached OpenAI's API result. Removing this step might sigificantly increase the running time of all the following DSPy programs depending on your OpenAI account setup.\n",
+        "\n"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "gXLDcxrLPgBO"
+      },
+      "outputs": [],
+      "source": [
+        "# This cell sets up the cache (pre-computed OpenAI call results).\n",
+        "!rm -r gsm8k_cache || true\n",
+        "!rm -r dspy || true\n",
+        "!git clone https://github.com/Shangyint/gsm8k_cache.git\n",
+        "\n",
+        "import os\n",
+        "repo_clone_path = '/content/gsm8k_cache'\n",
+        "\n",
+        "# Check if '/content' is writable\n",
+        "if not os.access('/content', os.W_OK):\n",
+        "    # If '/content' is not writable, choose an alternative directory\n",
+        "    # Example: using a directory relative to the current working directory\n",
+        "    repo_clone_path = os.path.join(os.getcwd(), 'gsm8k_cache')\n",
+        "\n",
+        "# Set up the cache for this notebook\n",
+        "os.environ[\"DSP_NOTEBOOK_CACHEDIR\"] = repo_clone_path"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "BiS0I8Oc9HDC"
+      },
+      "source": [
+        "Before we start, we install DSPy and all dependencies."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "u8WPRleVRkln"
+      },
+      "outputs": [],
+      "source": [
+        "%load_ext autoreload\n",
+        "%autoreload 2\n",
+        "\n",
+        "import sys\n",
+        "import os\n",
+        "import regex as re\n",
+        "import json\n",
+        "\n",
+        "import pkg_resources # Install the package if it's not installed\n",
+        "if not \"dspy-ai\" in {pkg.key for pkg in pkg_resources.working_set}:\n",
+        "    !pip install git+https://github.com/stanfordnlp/dspy.git@accenture-course\n",
```

Comment:
does this notebook need to install from this branch? I believe the latest changes here have been merged to main